src/libraries/Pool.sol

```diff
@@ -317,8 +319,16 @@ library Pool {
         state.feeGrowthGlobalX128 = zeroForOne ? self.feeGrowthGlobal0X128 : self.feeGrowthGlobal1X128;
         state.liquidity = cache.liquidityStart;
 
-        swapFee =
-            cache.protocolFee == 0 ? slot0Start.lpFee : uint24(cache.protocolFee).calculateSwapFee(slot0Start.lpFee);
+        // if the beforeSwap hook returned a valid fee override, use that as the LP fee, otherwise load from storage
+        uint24 lpFee;
+        if (!params.lpFeeOverride.isOverride()) {
+            lpFee = slot0Start.lpFee;
+        } else {
+            lpFee = params.lpFeeOverride.removeOverrideFlag();
+            if (!lpFee.isValid()) lpFee = slot0Start.lpFee;
```

Comment:
do you think we shoulda ctually just revert here? like should this be `lpFee.validate()` and if the hook returned an incorrect LP fee then thats an error and should revert?

Reply:
do you have a test for this case? where the hook returns a value >max?

Reply:
found it 😂